src/argilla/monitoring/_flair.py

```diff
@@ -63,7 +65,7 @@ def predict(self, sentences: Union[List[Sentence], Sentence], *args, **kwargs):
         metadata = kwargs.pop("metadata", None)
         result = self.__model__.predict(sentences, *args, **kwargs)
 
-        if isinstance(sentences, Sentence):
+        if not isinstance(sentences, list):
```

Comment:
Be aware of using this condition only will match for `list`. If you're using a generator or other kinds of sequences to provide a sentence batch, the `predict` wrapper will fail. Even if the original method (`model.predict`) is working

```python
instance((i for i in [1,2,3]), list) # False
```

Reply:
Even though you're right, I was following the type-hint for the `sentences` arg, that's why I kept that check as `isinstance(..., list)`, as I guess we're only expecting either a list of `Sentence` or a single `Sentence`.